### PR TITLE
Merge consequent open decls and hash directives into single structure blocks

### PR DESCRIPTION
**Pull request by vasily-kirichenko**
This fixes https://youtrack.jetbrains.com/issue/RIDER-10324 

/cc @auduchinok 

image

image

### CODE REVIEW

**Comment by vasily-kirichenko** [comment]
@dotnet-bot test Ubuntu14.04 Release_fcs Build please

File: src/fsharp/vs/ServiceStructure.fs [hunk 1]
```diff
@@ -545,7 +545,9 @@ module Structure =
                     | [], [] -> List.rev res
                     | [], _ -> List.rev (currentBulk::res)
                     | r :: rest, [] -> loop rest res [r]
-                    | r :: rest, last :: _ when r.StartLine = last.EndLine + 1 ->
+                    | r :: rest, last :: _ 
+                        when r.StartLine = last.EndLine + 1 || 
+                             sourceLines.[last.EndLine..r.StartLine - 2] |> Array.forall System.String.IsNullOrEmpty ->
```
Comment (saul) [inline src/fsharp/vs/ServiceStructure.fs]: Are we better off with IsNullOrWhitespace here?

Comment (vasily-kirichenko) [reply src/fsharp/vs/ServiceStructure.fs]: omg yes, I was sure I use this function :( Thanks for the review. Fixed.

**Comment by dsyme** [comment]
LGTM. Add a test to tests/service please

**Comment by vasily-kirichenko** [comment]
@dsyme done.

**Comment by dsyme** [comment]
@vasily-kirichenko Wow, thank you for such extensive tests, those are fantastic.  Very impressive

**Pull request merged**